MatrixSDK/MXRestClient.m

```diff
@@ -3966,21 +4018,41 @@ - (MXHTTPOperation*)requestEmailValidation:(NSString*)email
                                          }];
 }
 
-- (MXHTTPOperation *)submitEmailValidationToken:(NSString *)token
-                                   clientSecret:(NSString *)clientSecret
-                                            sid:(NSString *)sid
-                                        success:(void (^)())success
-                                        failure:(void (^)(NSError *))failure
+
+
+- (MXHTTPOperation *)submit3PIDValidationToken:(NSString *)token
+                                        medium:(NSString *)medium
+                                  clientSecret:(NSString *)clientSecret
+                                           sid:(NSString *)sid
+                                       success:(void (^)())success
+                                       failure:(void (^)(NSError *))failure
 {
+    NSString *path;
+    
+    if ([medium isEqualToString:kMX3PIDMediumEmail])
```

Comment:
I am not sure we need to limit the validation process to email and msisdn. If a new medium is created, we will need to deliver a new sdk.
Testing medium.count should be enough as sanity check.